MultiSlider.js

```diff
@@ -1,534 +1,455 @@
-    this.stepLength = this.props.sliderLength / this.optionsArray.length;
-
-    var initialValues = this.props.values.map(value =>
-      valueToPosition(value, this.optionsArray, this.props.sliderLength));
-
-    this.state = {
-      pressedOne: true,
-      valueOne: this.props.values[0],
-      valueTwo: this.props.values[1],
-      pastOne: initialValues[0],
-      pastTwo: initialValues[1],
-      positionOne: initialValues[0],
-      positionTwo: initialValues[1],
+    static propTypes = {
+        values: PropTypes.arrayOf(PropTypes.number),
+
+        onValuesChangeStart: PropTypes.func,
+        onValuesChange: PropTypes.func,
+        onValuesChangeFinish: PropTypes.func,
+
+        sliderLength: PropTypes.number,
+        touchDimensions: PropTypes.object,
+
+        customMarker: PropTypes.func,
+
+        customMarkerLeft: PropTypes.func,
+        customMarkerRight: PropTypes.func,
+        isMarkersSeparated: PropTypes.bool,
+
+        min: PropTypes.number,
+        max: PropTypes.number,
+        step: PropTypes.number,
+
+        optionsArray: PropTypes.array,
+
+        containerStyle: ViewPropTypes.style,
+        trackStyle: ViewPropTypes.style,
+        selectedStyle: ViewPropTypes.style,
+        unselectedStyle: ViewPropTypes.style,
+        markerContainerStyle: ViewPropTypes.style,
+        markerStyle: ViewPropTypes.style,
+        pressedMarkerStyle: ViewPropTypes.style,
+        valuePrefix: PropTypes.string,
+        valueSuffix: PropTypes.string,
+        enabledOne: PropTypes.bool,
+        enabledTwo: PropTypes.bool,
+        onToggleOne: PropTypes.func,
+        onToggleTwo: PropTypes.func,
+        allowOverlap: PropTypes.bool,
+        snapped: PropTypes.bool,
+        markerOffsetX: PropTypes.number,
+        markerOffsetY: PropTypes.number,
+        vertical: PropTypes.bool
     };
-  }
-
-  componentWillMount() {
-    var customPanResponder = (start, move, end) => {
-      return PanResponder.create({
-        onStartShouldSetPanResponder: (evt, gestureState) => true,
-        onStartShouldSetPanResponderCapture: (evt, gestureState) => true,
-        onMoveShouldSetPanResponder: (evt, gestureState) => true,
-        onMoveShouldSetPanResponderCapture: (evt, gestureState) => true,
-        onPanResponderGrant: (evt, gestureState) => start(),
-        onPanResponderMove: (evt, gestureState) => move(gestureState),
-        onPanResponderTerminationRequest: (evt, gestureState) => false,
-        onPanResponderRelease: (evt, gestureState) => end(gestureState),
-        onPanResponderTerminate: (evt, gestureState) => end(gestureState),
-        onShouldBlockNativeResponder: (evt, gestureState) => true,
-      });
+
+    static defaultProps = {
+        values: [0],
+        onValuesChangeStart: () => {},
+        onValuesChange: values => values,
+        onValuesChangeFinish: values => values,
+        step: 1,
+        min: 0,
+        max: 10,
+        touchDimensions: {
```

Comment:
`styles`

Reply:
Those aren't really styles, `slipDisplacement` is used to calculate position for example